easy-ui-react/src/ProductLayout/ScreenSizeRenderer.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { ReactNode, useEffect } from "react";
+import { productLayoutMediaQuery } from "./constants";
+
+import styles from "./ScreenSizeRenderer.module.scss";
+
+export type ScreenSizeRendererProps = {
+  onChange?: (isDesktop: boolean) => void;
+  renderOnLargeScreen: () => ReactNode;
+  renderOnSmallScreen: () => ReactNode;
+};
+
+/**
+ * SSR-compatible way to render different component trees whether the product
+ * layout is on a small screen or a large screen.
+ *
+ * It does this by rendering both component trees and using media queries
+ * through CSS to remove the applicable tree from the user agent (using
+ * display: none).
+ *
+ * Provides a change listener for adjusting component state as needed.
+ *
+ * @private
+ * @ignore
+ */
```

Comment:
the idea is to use this in as isolated a manner as possible as a way to render more complex trees across small/large screens instead of trying to reorganize elements across screen sizes, which can get complicated/messy.

the downside to this is having multiple nodes in the document tree, which could cause potential confusion to scripts that may try to target specific nodes (i.e. walkme?), even though they won't be exposed to user agents with display: none; set. not sure this would be an actual issue but may be something to keep an eye on

Reply:
makes sense and i agree with this approach. do you anticipate we'll ever want/need to use this outside of layouts? im wondering if something like `LayoutScreenSizeRenderer` would be a bit more explicit about not just what it does, but also the purpose of where/how we intend to use this (sparingly). although the function itself is generic, so im not opposed to the current name either

Reply:
i agree with that. i renamed it to `LayoutScreenSizeRenderer`